compiler/src/dmd/opover.d

```diff
@@ -1149,7 +1149,31 @@ private Expression compare_overload(BinExp e, Scope* sc, Identifier id, ref EXP
     if (e.isEqualExp() && ad1 == ad2)
         return null;
     Expression result = checkAliasThisForLhs(ad1, sc, e, aliasThisStop);
-    return result ? result : checkAliasThisForRhs(isAggregate(e.e2.type), sc, e, aliasThisStop);
+    if (result)
+        return result;
+
+    result = checkAliasThisForRhs(isAggregate(e.e2.type), sc, e, aliasThisStop);
+    if (result)
+        return result;
+
+    if (s || s_r)
+        return null;
+
+    Expression suggestOverloading(Expression other, AggregateDeclaration ad)
+    {
+        error(e.loc, "no operator `%s` for type `%s`", EXPtoString(e.op).ptr, ad.toChars);
+        string op = e.isEqualExp() ? "bool" : "int";
+        errorSupplemental(ad.loc, "perhaps overload it with `%.*s %s(%s other) const {}`", op.fTuple.expand, id.toChars, other.type.toChars);
```

Comment:
This introduced a regression @dkorpel 
```
dmd/opover.d:1144:94: error: no property ‘fTuple’ for ‘op’ of type ‘string’
```
For reasons I can't fathom, can't reproduce this with dmd 2.103.x, but seen with all releases of gdc-(9..14)

The error doesn't look wrong either - there's no `import dmd.root.string;` anywhere in the opover module!

Reply:
I added a `public import dmd.root.string: fTuple;` to dmd/errors.d, right below the `public import core.stdc.stdarg;`. Are selective public imports broken on gdc?

Reply:
Ah, you added `public import` to the errors module - action at a spooky distance. :-)

I didn't bother syncing this as gdc implementation of dmd.errors is just a set of forward declarations.